### DISCUSSION TITLE
Fitsnap/Dakota-Optimization of group weights

### DISCUSSION THREAD

**Discussion by Rana-Phy** [FitSNAP/FitSNAP#89] [Ideas]
Dear Developers,

Greetings. I am trying to fit snap hyperparameters and group weights using Dakota.
In Fitsnap papers I am seeing those hyperparameters (cut and weight of elements) have been optimized first  (by setting equal weights of all groups?) then group weights have been optimized. 
Since we have a lot of groups to optimize it will take a long time to fit.

Hope is in the following document it is saying that
https://www.osti.gov/servlets/purl/1641365

**Shorter time during groupweight fitting since we use previous Data and DumpSnap files so we don't have to convert JSON to LAMMPS data files or run LAMMPS again**

Could you kindly share how to read previous data by Fitsnap?

Best regards,
Rana
(rana.phy.buet@gmail.com)

**Comment by mitwood** [comment]
Hey @Rana-Phy,
What do you mean previous data read in? In those papers we would first optimize the radial cutoff and element density weights (wj), both found in the [BISPECTRUM] part of the input file. These terms directly affect the calculation of the descriptors (the group weight do not), and thus the resultant fit is sensitive to small changes in these terms. While optimizing these, we are generally using smartweights = 1, and 1:1:1 weights of energy:force:stress. Following this hyper parameter optimization, the radial cutoff and element density terms are fixed in the input file while DAKOTA optimizes the group weights.
-Mitch

**Reply by Rana-Phy** [reply]
Dear Dr. @mitwood ,

Thanks for your reply.
I am looking at the following link.
https://github.com/FitSNAP/FitSNAP/tree/22Mar18/examples/Dakota

As you said after optimizing cutoff and weight for each element with fixed smartweights = 1, and 1:1:1 weights of energy: force: stress, I need to optimize group weights. And here we have many free variables.
During optimization of group weights, cutoff, and element density weights will be fixed.

In the latest version of fitsnap is it possible to write and read 'DumpSnap and data' ? It may reduce fitting time.
Or I should go for 22Mar18 version.

Best regards,
Rana

**Reply by mitwood** [reply]
So that version is very outdated and I would highly suggest to use the current master branch. The old style of outputting Data and DumpSnap as part of the fit was replaced with real-time calls to lammps to gather the descriptors, which we found to be faster(since these are very small configurations that LAMMPS processes) than the time it takes to read from disk the numerous files needed to restart a fit.

That said, we are working on different ways to restart a fit by outputting the large matrices of descriptors and ground truth values. In the meantime, try to test the speed of the modern FitSNAP versions.

-Mitch
